src/repo.rs

```diff
@@ -207,52 +226,53 @@ impl ArcWake for RepoWaker {
 }
 
 /// The backend of a repo: the repo runs an event-loop in a background thread.
-pub struct Repo<T: NetworkAdapter> {
+pub struct Repo {
     /// The Id of the repo.
     repo_id: RepoId,
 
-    network_adapter: Box<T>,
+    network_adapters: HashMap<RepoId, Box<dyn NetworkAdapter<Error = NetworkError>>>,
     documents: HashMap<DocumentId, DocumentInfo>,
 
     stream_waker: Arc<RepoWaker>,
     sink_waker: Arc<RepoWaker>,
 
     /// Messages to send on the network adapter sink.
-    pending_messages: VecDeque<NetworkMessage>,
+    pending_messages: HashMap<RepoId, VecDeque<NetworkMessage>>,
     /// Events received on the network stream, pending processing.
     pending_events: VecDeque<NetworkEvent>,
 
     /// Callback on applying sync messages.
-    sync_observer: Option<Box<dyn Fn(Vec<DocumentId>) + Send>>,
+    sync_observer: Option<Box<dyn Fn(Vec<DocHandle>) + Send>>,
```

Comment:
Note: changed this to passing doc handles to the observer, because in the current workflow we start syncing when a new peer connects, even if the local client does not have an existing doc handle. So there is a need for a way to get a doc handle to such a doc that is synced by the "backend only", and this is the current way.